### PR TITLE
Travis: osx and linux build

### PR DESCRIPTION
**Pull request by jburel**
Run ``gradle build on linux`` and ``gradle packageApplicationDmg`` on mac

* insight dmg will push to GH when a tag is set
* importer dmg will be built and pushed to GH when a tag is set

This has been tested using merge build of this PR and #22 https://github.com/jburel/omero-insight/releases/tag/v5.5.0-m14
* Check that travis is green
* Download the dmg and check that it can start

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -7,11 +10,16 @@ cache:
     - $HOME/.gradle/caches/
     - $HOME/.gradle/wrapper/
 
+before_install:
+  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then brew update          ; fi
```
Comment (joshmoore) [inline .travis.yml]: FYI: The osx build ended up taking 6x the linux time. 6 of the 12 minutes were this all.

Comment (jburel) [reply .travis.yml]: I tried previous builds and it was the same time

Comment (jburel) [reply .travis.yml]: It will always take longer to build a dmg or exe, not specific to travis, the same is true locally

**Comment by jburel** [comment]
I have tested the experimental windows support, but it is not ready yet.

**Comment by manics** [comment]
Tested with this change to travis.yml:
```diff
diff --git a/.travis.yml b/.travis.yml
index 7e458a46b..273e05006 100644
--- a/.travis.yml
+++ b/.travis.yml
@@ -20,6 +20,9 @@ install:
 script:
   - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then ./gradlew packageApplicationDmg ; fi
   - if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then ./gradlew build ; fi
+  - ls -l build/distributions/*
+  - test -f build/distributions/OMERO.imagej-*
+  - test -f build/distributions/OMERO.insight-*

 deploy:
   provider: releases
```
Linux worked https://travis-ci.com/manics/omero-insight/jobs/197137782
```
$ ls -l build/distributions/*
-rw-rw-r-- 1 travis travis 144306936 May  2 11:02 build/distributions/OMERO.imagej-5.5.0-SNAPSHOT.zip
-rw-rw-r-- 1 travis travis 144312210 May  2 11:02 build/distributions/OMERO.insight-5.5.0-SNAPSHOT.zip
The command "ls -l build/distributions/*" exited with 0.
$ test -f build/distributions/OMERO.imagej-*
The command "test -f build/distributions/OMERO.imagej-*" exited with 0.
$ test -f build/distributions/OMERO.insight-*
The command "test -f build/distributions/OMERO.insight-*" exited with 0.
```
OS X didn't: https://travis-ci.com/manics/omero-insight/jobs/197137781
```
$ ls -l build/distributions/*
ls: build/distributions/*: No such file or directory
The command "ls -l build/distributions/*" exited with 1.
$ test -f build/distributions/OMERO.imagej-*
The command "test -f build/distributions/OMERO.imagej-*" exited with 1.
$ test -f build/distributions/OMERO.insight-*
The command "test -f build/distributions/OMERO.insight-*" exited with 1.
```

**Comment by jburel** [comment]
Your tests are not valid for mac since ``build/distributions/`` is not created when ``gradle packageApplicationDmg`` is run. A packaged folder is created then the dmg is under ``packaged/main/bundles``
I will improve the travis file

File: .travis.yml (hunk 1 above)
Comment (jburel) [reply .travis.yml]: Another reason it slower on mac is that the image does not come with gradle, the linux one does.
So extra installation time

**Comment by manics** [comment]
`deploy.file: build/distributions/*` will need to be updated then to ensure the OS X artefacts can be released.

**Comment by jburel** [comment]
last commit tested on my account https://github.com/jburel/omero-insight/releases/tag/v5.5.0-m10 (tag number is irrelevant)

**Comment by jburel** [comment]
I have just tested the dmg generated by travis then pushed to gh. It is not usable 
Several things are missing

**Comment by will-moore** [comment]
Insight from https://github.com/jburel/omero-insight/releases/download/v5.5.0-m14/OMERO.insight-5.5.0-SNAPSHOT.dmg starts and works fine. Same for Importer.

**Comment by will-moore** [comment]
Travis was green at https://github.com/ome/omero-insight/pull/23/commits/edd9aa802912a835857f5441f3ae74bf6aa25a9f but hasn't run against the 2 subsequent commits?

**Comment by jburel** [comment]
For some reasons, travis does not want to restart the check. 
The changes were also tested on another branch. I will adjust the travis file in a follow-up PR if required.

**Comment by jburel** [comment]
Ok I find out the issue, it was an indentation problem. Travis was not parsing the request. i will push a commit when travis is green on my account

**Pull request merged**